src/lib.rs

```diff
@@ -57,21 +52,22 @@ fn pattern_to_glob(pattern: impl AsRef<str>) -> Result<CodeownersGlob, ParseErro
         pattern = format!("**/{}", pattern);
     }
 
-    Ok(CodeownersGlob {
-        glob: globset::Glob::new(&pattern)?,
-        absolute,
-    })
+    pattern
 }
 
 /// Representation of one Codeowner pattern and the respective list of owners.
 #[derive(Debug)]
 pub struct Rule {
     pub pattern: String,
     pub owners: Vec<Owner>,
+    pub matcher: globset::GlobMatcher,
 }
 
 impl PartialEq for Rule {
     fn eq(&self, other: &Rule) -> bool {
+        /* We purposefully don't want to compare the glob, as it is
```

Comment:
That is what I had  this here before :D I ripped out the matcher from the Rule for some reason...